merlin/systems/dag/ops/operator.py

```diff
@@ -120,6 +123,39 @@ def create_node(self, selector: ColumnSelector) -> InferenceNode:
         """
         return InferenceNode(selector)
 
+    @classmethod
+    def from_model_registry(cls, registry: ModelRegistry, **kwargs) -> "InferenceOperator":
+        """
+        Loads the InferenceOperator from an MLflow Model Registry.
+
+        This only works for operators that extend this base class and accept a model path
+        as an argument to its __init__ function, such as PredictTensorflow. A counter example would
+        be TransformWorkflow, which does not load a model.
+
+        Parameters
+        ----------
+        registry : ModelRegistry
+            A ModleRegistry object that will provide the path to the model.
+        **kwargs
+            Other kwargs to pass to your InferenceOperator's constructor.
+
+        Returns
+        -------
+        InferenceOperator
+            New node for Ensemble graph.
+        """
+
+        # validate that we have a model_or_path argument. This is a total hack and perhaps a smell
+        # that all inference operators should accept a model_or_path, or at least we introduce
+        # another layer of base class that does.
```

Comment:
One trick I use sometimes when there are two options is to make one of them the type expected by the constructor and then add a class method that takes the other type and converts it. In this case, I'd probably pass the model type into constructors, and add a class method like `from_path` that loads the models and passes them to the constructors.

Reply:
That's a great idea. In the case of `TransformWorkflow` the `from_path` method could throw some kind of exception since it doesn't really make sense for that operator.

Reply:
I think it would make sense if you want to load a Workflow from a file, which is not an uncommon thing to do.